### PR TITLE
feat(preset)!: decrease prConcurrentLimit to 10

### LINKED ISSUE #11909: Decrease `:prConcurrentLimit` from 20 to 10 in our `config:base` preset
### What would you like Renovate to be able to do?

We recently changed our documentation to recommend that most people set their `prConcurrentLimit` to something lower than the default of 20 that comes with our `config:base` present.

It might be good to change the `config:base` preset so that end-users follow this new recommendation automatically.

See this comment from the head maintainer: https://github.com/renovatebot/renovate/pull/11879#discussion_r715500459

### If you have any ideas on how this should be implemented, please tell us here.

We already have a `:prConcurrentLimit10` preset that we could extend from in our `config:base` preset.
We could change this line here to use that lower limit:

https://github.com/renovatebot/renovate/blob/42bef1e9398e5ef87a883ac2dd2755a0ec679e20/lib/config/presets/internal/config.ts#L12

I tried doing this without changing anything else, and it breaks the Jest tests, so we probably need to do a bit more than just changing this line... 😄 😉 

We would also need to change the documentation again, as we mention the number 20 in this file: https://github.com/renovatebot/renovate/blob/main/docs/usage/configuration-options.md

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by olegkrivtsov**
## Changes:

Decrease :prConcurrentLimit from 20 to 10 in our config:base preset #11909

## Context:

Closes #11909

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
@rarkins We need to prepare a new major version branch.

File: docs/usage/configuration-options.md
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:246]: Do we really mean for people to put `branchConcurrentLimit` as low as `3`? I imagine setting it to `5` will be low enough already? Is there any need to go lower than `5`?

Where does this number `3` come from anyway? I don't see anything in this PR that explains that number, what am I missing here?

Comment (olegkrivtsov) [reply docs/usage/configuration-options.md:246]: Hi @HonkingGoose I used value 3 because the old text was "values as 5 or 10". I replaced it with "values 5 or 3", because 10 is already used by default now. What value do you recommend to use?

Comment (rarkins) [inline docs/usage/configuration-options.md:246]:
```suggestion
The `config:base` preset that many extend from limits the amount of concurrent branches to 10, but in many cases a limit as low as 3 or 5 can be most efficient for a repository.
```

Comment (HonkingGoose) [reply docs/usage/configuration-options.md]: I see that the head maintainer already made a suggestion that applies to this line. I think this is okay now. Somebody with `write` rights can dismiss my review. 😉

**Review by viceice — APPROVED** [review]

**Review by HonkingGoose — APPROVED** [review]

**Pull request merged**
